### PR TITLE
HELM-369: Flow datasource and histogram panel code improvements

### PR DESCRIPTION
**Pull request by synqotik**
A number of code improvements related to the Flow Datasource and Flow Histogram panel which were requested on PR #597 , including:

- breaking functions down into smaller chunks
- removing `moment.js` reference and `eslint-disable` clause
- type annotations
- other code cleanup and comments

# External References

* JIRA (Issue Tracker): https://opennms.atlassian.net/browse/HELM-369
* Continuous Integration: [CircleCI](https://circleci.com/gh/OpenNMS/grafana-plugin)

### CODE REVIEW

File: src/datasources/flow-ds/helpers.ts [hunk 1]
```diff
@@ -716,36 +736,29 @@ const processRawSeriesData = (queryData: FlowParsedQueryRow, options: FlowQueryR
...
-    // so modifying here to understand what data should be received
-    const metric = queryRow.segment ? queryRow.segment.label : undefined;
-    const fields: any[] = []
-
-    const dataFrame = {
-        name: queryRow.refId,
-        fields: fields,
-        meta: toBits ? { custom: { metric, toBits: '' } } : { custom: { metric } }
-    }
+    const dataFrame = createSummaryDataFrame(queryRow, rows.length)
```
Comment (scthompsonopen) [inline src/datasources/flow-ds/helpers.ts:751]: Amazing. This makes me happy.

File: src/panels/flow-histogram/FlowHistogramHelpers.ts [hunk 2]
```diff
@@ -1,44 +1,52 @@
...
-        }
-        case 'Stacked': {
-            return processedData.stackedData?.metricsByLabel ? processedData.stackedData.metricsByLabel : []
-        }
-        default: return []
-    }
+  const mode = options.flowHistogramOptions.mode.label
+
+  if (mode === 'Separate') {
+    return createSeparateFlotDataSeries(processedData, options.flowHistogramOptions.direction.label)
```
Comment (scthompsonopen) [inline src/panels/flow-histogram/FlowHistogramHelpers.ts:15]: More smiles! This is a great PR.

File: src/panels/flow-histogram/FlowHistogramHelpers.ts [hunk 3]
```diff
@@ -79,142 +87,146 @@ export const getFlowHistogramPlotConfig = (processedData: FlowPanelDataProcessed
...
+    separateData: {
+      inByLabel: inResult,
+      outByLabel: outResult,
+      ticks: indexedMetricLabels
+    },
+    stackedData: {
+      metricsByLabel: stackedResult,
+      ticks: [[FlowDataDirection.dataIn.value, FlowDataDirection.dataIn.label], [FlowDataDirection.dataOut.value, FlowDataDirection.dataOut.label]]
     }
+  }
```
Comment (scthompsonopen) [inline src/panels/flow-histogram/FlowHistogramHelpers.ts:145]: Great stuff in here too. Nice cleanups.

**Review by scthompsonopen — APPROVED** [review]
Looks incredible. Great work. :)

**Comment by synqotik** [comment]
> Looks incredible. Great work. :)

Thank you! It was good to finally follow up on all your great suggestions :)

**Pull request merged**